go.mod

```diff
@@ -6,6 +6,7 @@ require (
 	github.com/evanphx/json-patch v4.9.0+incompatible
 	github.com/go-logr/logr v0.4.0
 	github.com/google/go-cmp v0.5.4
+	github.com/google/uuid v1.1.2
```

Comment:
No, it's a relatively easy swap out -- actually since apimachinery's uuid is mostly focused on types.UID (which is itself a string type), using that and just making sure everything is a string makes some things cleaner. (I didn't see us use any particular UUID package, though I feel like this has come up before...